Src/Common/StringUtilities.cs

```diff
@@ -7,7 +7,12 @@
     /// <summary>
     /// Generic functions to perform common operations on a string.
     /// </summary>
-    public static class StringUtilities
+#if DEPENDENCY_COLLECTOR
+    public
```

Comment:
I like hiding this class from public surface. What's the reason to keep it public?

Reply:
this class is used by AspNetCore SDK and should be visible to it. There are two alternatives:
- copy paste (as it is done now and i'm getting read of it by this PR)
- move common code to based SDK. I'm not fond of it because most of W3C code will go away from AI SDK eventually and I'd prefer to keep it all in one place. We can move other StringUtilities common methods in different PR